### ISSUE TITLE
Support for full color palette

### ISSUE THREAD

**Issue by ad-si** [quchen/prettyprinter#65]
Why is there only support for 16 colors?
The full palette would be cool! :grin:

![screen shot 2018-07-06 at 17 10 58](https://user-images.githubusercontent.com/36796532/42391641-95873de6-813f-11e8-8ce1-9745224d89e5.png)

**Comment by quchen** [comment]
There’s not enough supply or demand for this feature. ;-)

**Comment by ad-si** [comment]
😭

**Comment by quchen** [comment]
You could easily implement this though if you needed it! Just extend [`data Color`](http://hackage.haskell.org/package/prettyprinter-ansi-terminal-1.1.1.2/docs/Data-Text-Prettyprint-Doc-Render-Terminal-Internal.html#t:Color) by a field `RGB Word8 Word8 Word8` and follow the compiler warnings to implement it. `ansi-terminal` has support for RGB SGR codes [(link)](http://hackage.haskell.org/package/ansi-terminal-0.9.1/docs/System-Console-ANSI-Types.html#t:SGR), so all in all this should not be a hard thing to implement.

I’m very much open to a pull request if you decided to implement this, but at the moment I’m afraid I don’t have the time myself.

**Comment by afcowie** [comment]
Would you still be amenable to a patch in this direction? I just asked the same question and came to the same https://hackage.haskell.org/package/ansi-terminal-0.10.1/docs/System-Console-ANSI-Types.html#v:SetRGBColor constructor which perhaps we can pass through instead? (I note that it's `Colour Float` there, which is interesting)

**Comment by quchen** [comment]
Yes, I’m open to it! I’d like to keep the feature small though, without too many color management bells and whistles.

**Comment by spacekitteh** [comment]
If @afcowie hasn't got the patch done, I'll do one over the next couple of days.

**Comment by afcowie** [comment]
@spacekitteh haven't had a chance yet, sorry.

**Comment by sjakobi** [comment]
Reopening due to the renewed interest.

**Comment by quchen** [comment]
Re-closing due to renewed disinterest ;-)

A PR implementing this would still be welcome though.

**Comment by sjakobi** [comment]
Re-re-opening since there is a PR: #224.
